### DISCUSSION TITLE
Teleport Community Edition will restrict commercial usage starting with version 16

### DISCUSSION THREAD

**Discussion by klizhentas** [gravitational/teleport#39158] [General]
We open sourced Teleport in 2015 with a mission to secure access to infrastructure. It has since become a popular [open source project](https://github.com/gravitational/teleport) with over 15,000 GitHub stars and is licensed under [AGPLv3](https://opensource.org/licenses/AGPL-3.0).

We have been offering the stable releases of the Teleport Community Edition binaries and images distributed under the permissive [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0) license. 

Starting with the June release of Teleport 16, we are switching Teleport Community Edition to a commercial license with new restrictions:

* Individuals are free to use Teleport Community Edition for personal and hobby use with no restrictions.
* **Companies may use Teleport Community Edition on the condition they have less than 100 employees __and__ less than $10MM in annual recurring revenue (ARR).**
* Companies cannot resell or embed Teleport Community Edition in their products or services.
* Companies larger than 100 employees or more than $10MM ARR need to contact our sales team to evaluate and use Teleport. 
* We will stop distributing compiled binaries, container images, and Amazon Machine Images (AMIs) under Apache 2.0 starting with Teleport 16. We will switch the license of those assets to the commercial license of Teleport Community Edition.
* We will continue to support previous versions according to our [published schedule](https://goteleport.com/docs/faq/#which-version-of-teleport-is-supported). Additionally, we will maintain the Apache 2.0 license for Community Edition binaries, containers, and images for Teleport Community Edition versions **prior to** Teleport 16.

Please read more details in our [blog post announcement](https://goteleport.com/blog/teleport-community-license/).

**Comment by buzzzo** [comment]
No good news but... what does it means from a pratical point of view: if i upgrade the community version 15.x to 16.x , what happens ? does it works as before or do i need to register in order to prove i'm an homelabber or company that fits in the license requirements ?

**Reply by webvictim** [reply]
When you start version 16, there is a disclaimer on the sign-in screen asking you to self-certify that you're eligible to use Teleport Community edition.

**Reply by buzzzo** [reply]
OK so currently it's just a disclaimer and you trust that who are using the software is telling the truth by "self certify" its usage.
So the "self certification" it's just an assumption not something that need to be proved in any way by the user ?

**Reply by webvictim** [reply]
Correct, it's an honour system.

**Reply by buzzzo** [reply]
Dunno if this is just the starting point for the way to close completely the software  but surely companies that does not fit the community license requirements will keep using teleport in "unethical"  licensed way until teleport company force them to prove they fit the license.

Would be better to plan a fair price for the "community" packed version (similar to what proxmox do).

Thx

**Comment by dkowis** [comment]
Does this mean that the community edition will have feature parity with the commercial edition? 

As a homelabber, I use Keycloak, not github auth, and that's restricted to the commercial edition.

**Reply by webvictim** [reply]
No, the feature split between Teleport Community and Teleport Enterprise remains the same as it is.

**Reply by buzzzo** [reply]
Just to understand: if I compile from source the community edition the license remains agpl so the resultant binaries will be not subjected to the new restricted license ?

**Reply by webvictim** [reply]
> Just to understand: if I compile from source the community edition the license remains agpl so the resultant binaries will be not subjected to the new restricted license ?

Correct, binaries that you compile from source yourself will be subject to the [AGPLv3 license](https://opensource.org/license/agpl-v3).

**Reply by dkowis** [reply]
I guess another not-so-sure question: 

If I build it myself, will I have access to OIDC authentication?

**Reply by webvictim** [reply]
> If I build it myself, will I have access to OIDC authentication

No. This is an Enterprise feature, only available with a commercial Teleport license.

**Reply by TwiggyWan** [reply]
> > Just to understand: if I compile from source the community edition the license remains agpl so the resultant binaries will be not subjected to the new restricted license ?
> 
> Correct, binaries that you compile from source yourself will be subject to the [AGPLv3 license](https://opensource.org/license/agpl-v3).

> We will stop distributing compiled binaries, container images, and Amazon Machine Images (AMIs) under Apache 2.0 starting with Teleport 16. We will switch the license of those assets to the commercial license of Teleport Community Edition.

Hi, so are we allowed to compile teleport without modification, and then distribute binaries & images free of charge ourselves?

**Reply by webvictim** [reply]
You are allowed to compile Teleport yourself and distribute these binaries and images under the terms of the [AGPLv3 license](https://opensource.org/license/agpl-v3).

**Reply by p0358** [reply]
I think they cannot change the baseline software source code license since it's AGPLv3 and they'd need permission from all 338 contributors to do that. So if I understand correctly, they *maybe* can place additional restrictions like they did on the Community binaries/package distributions, but they still have to abide by the AGPLv3 terms, including keeping the code open.

But I wonder how this pertains to the GPL freedoms:
> Nobody should be restricted by the software they use. There are four freedoms that every user should have:
>
>    the freedom to use the software for any purpose,
>    the freedom to change the software to suit your needs,
>    the freedom to share the software with your friends and neighbors, and
>    the freedom to share the changes you make.

I know that GPL doesn't prevent things like selling your software, as long as you still provide the source code to people who buy it, in order for them to exercise said freedoms. And technically it is always within your rights to also recompile the software to remove any imposed limitations...

**Reply by buzzzo** [reply]
The sources remain under agpl but you need to compile them and hope that all the build system remain well documented and maintained.

Unless someone starts something like centos (take rhel sources and build binary ) it's hard to stay on teleport for free .

**Comment by compgeniuses** [comment]
just when i was assessing implementing teleport for my company then boom this happens.

I think ill just wait until i can afford the enterprise license then

**Comment by dbodow** [comment]
In the linked blog post:

> Why are we making this change?
> Teleport is a for-profit business, and we are spending an increasing amount of time and expense maintaining the free Teleport Community Edition.
> 
> We are seeing growing interest and support requests in our community discussions forums from individuals and organizations.
> 
> There are thousands of people in our Slack channel with an ever-growing volume of support requests coming every day, many of them coming from companies that are many times our size.
> 
> We love helping out small teams and hobbyists with securing their environments, but we are not in a position to support large companies for free.

This doesn't hold up at all. Just require an enterprise contract for Slack support and issue logging; you can absolutely release and maintain a community edition without using any resources whatsoever on supporting individual users of it. I used to recommend self-hosted CE Teleport to friends; I am deeply embarrassed that I have done so now.

This is clearly a bait-and-switch where Gravitational has tried to get users to lock into their ecosystem and then charge costly fees without any warning. What a terrible foot to start new business relationships on!

**Reply by DocterDum** [reply]
If this was targeting businesses with >$100,000 ARR then I might agree, but it's businesses with >$10MM - A factor of 100 higher...
Also, the source code itself and all self-compiled binaries are still AGPL...

> > Just to understand: if I compile from source the community edition the license remains agpl so the resultant binaries will be not subjected to the new restricted license ?
> 
> Correct, binaries that you compile from source yourself will be subject to the [AGPLv3 license](https://opensource.org/license/agpl-v3).

Long story short, this doesn't apply to you. And on the off chance it does apply to you, you should already have some method in place to be contributing to the project.

**Comment by 3uzbcqje** [comment]
Does anyone know if there's a fork of version 15 that the community is going to maintain? Something similar to how OpenTofu is the new FOSS version of Terraform.

**Comment by YusufRamlawi** [comment]
@klizhentas 
if we have (big) less than $10M ARR but more than 100 employees, about 40 actual users. can or cannot user the Community Edition?

**Reply by webvictim** [reply]
> Companies may use Teleport Community Edition on the condition they have less than 100 employees and less than $10MM in annual recurring revenue (ARR).

As you have more than 100 employees, you would not qualify under the terms of the Community edition license.

**Comment by ElijahLynn** [comment]
Can you make the enterprise edition available via homebrew?

Easiest way to get updates is through `brew`, vs being a caveperson and downloading with curl. With your proposed way of installing, one has to manually check for updates, which feels less secure to me because I'm probably not going to remember to do it.

https://goteleport.com/docs/connect-your-client/tsh/

> Using Homebrew to install Teleport is not supported. The Teleport package in Homebrew is not maintained by Teleport and we can't guarantee its reliability or security.

Is the reason because of security? You kinda imply that but not explicitly. Brave browser distributes through Homebrew so surely it can be done securely, yah?

**Reply by ElijahLynn** [reply]
also, if anyone is getting this WARNING with the community edition:
```
2025/10/21 11:48:59 WARN Client tools updates are disabled as they are licensed under AGPL. To use Community Edition builds or custom binaries, set the 'TELEPORT_CDN_BASE_URL' environment variable.
```

You can export it to be `https://cdn.teleport.dev` and it will remove the Warning.
